### ISSUE TITLE
3.0 RC1: DateTimeType errors

### ISSUE THREAD

**Issue by rfenner** [cakephp/cakephp#5631]
While looking over the DateTimeType in database/Type I noticed what looked to be a typo and missing space in the format marshal function

at line 126 
```php
            (is_numeric($value['year']) & is_numeric($value['month']) && is_numeric($value['day']))

```
should probably be 
```php
            (is_numeric($value['year']) && is_numeric($value['month']) && is_numeric($value['day']))
```

For the missing space when adding in the time the code at line 135 is
```php
        $format .= sprintf('%02d:%02d:%02d', $value['hour'], $value['minute'], $value['second']);
```
thus the resulting formatted date would 'Y-m-dH:i:s.
a fix would be to add a space between the time and date if the date is there such as 
```php
        $format .= sprintf('%s%02d:%02d:%02d', (empty($format)  ? '' : ' '), $value['hour'], $value['minute'], $value['second']);
```

**Comment by markstory** [comment]
You can either submit the changes as a pull request, or I can take care of the changes tonight.

**Comment by rfenner** [comment]
It would probably be easier for you to make the changes as I'm not setup to push changes.

**Comment by markstory** [comment]
Cool, I'll take care of it tonight then. Thanks for pointing out the issues.

**Comment by markstory** [comment]
Fixed in c6367778ff643aba39110628b4e5672da9234850
